packages/next/src/elements/Nav/getNavPrefs.ts

```diff
@@ -1,36 +1,27 @@
-import type { NavPreferences, Payload, User } from 'payload'
+import type { NavPreferences, Payload, PayloadRequest, User } from 'payload'
 
 import { cache } from 'react'
 
 export const getNavPrefs = cache(
-  async ({ payload, user }: { payload: Payload; user: User }): Promise<NavPreferences> =>
+  async ({
+    payload,
+    req,
+    user,
+  }: {
+    payload: Payload
+    req: PayloadRequest
+    user: User
+  }): Promise<NavPreferences> =>
     user
       ? await payload
-          .find({
-            collection: 'payload-preferences',
-            depth: 0,
-            limit: 1,
-            user,
-            where: {
-              and: [
-                {
-                  key: {
-                    equals: 'nav',
-                  },
-                },
-                {
-                  'user.relationTo': {
-                    equals: user.collection,
-                  },
-                },
-                {
-                  'user.value': {
-                    equals: user.id,
-                  },
-                },
-              ],
+          .findPreferenceByKey({
+            key: 'nav',
+            req: {
+              ...(req || ({} as PayloadRequest)),
```

Comment:
I know the Node `req` had a bug if we spread it. That's where the `isolateObjectProperty` thing came into play - because we specifically cannot spread `req`. Not sure if it's the same with the Web Fetch API, but spreading `req` here probably is not necessary anyway. Let's make sure that [this PR](https://github.com/payloadcms/payload/pull/9511) makes it so that `user` is indeed required, which is proper, but the `req` is not.